doc/1/controllers/document/update/index.md

```diff
@@ -0,0 +1,125 @@
+| `id        `       | <pre>String</pre>                            | Document ID                        |
+| `document`         | <pre>ConcurrentHashMap<String, Object></pre> | Partial document content |
+| `options`          | <pre>UpdateOptions</pre><br>(`null`)         | Optional parameters               |
+
+---
+
+### options
+
+A [UpdateOptions](/sdk/java/3/core-classes/update-options) object.
+
+The following options can be set:
+
+| Arguments          | Type                                         | Description                       |
+| ------------------ | -------------------------------------------- | --------------------------------- |           |
+| `waitForRefresh`   | <pre>Boolean</pre>                           | If set to `true`, Kuzzle will wait for the persistence layer to finish indexing|
+| `retryOnConflict`  | <pre>Integer</pre>                           | The number of times the database layer should retry in case of version conflict |
+| `source`           | <pre>Boolean</pre>                           | If true, returns the updated document inside the response |
+
+## Return
+
+A `ConcurrentHashMap` which has the following properties:
+
+| Property     | Type                         | Description                                                    |
+|------------- |----------------------------- |--------------------------------------------------------------- |
+| `_source`    | <pre>ConcurrentHashMap</pre> | Updated document (If source option set to true)                |
+| `_id`        | <pre>String</pre>            | ID of the updated document                                     |
+| `_version`   | <pre>Integer</pre>           | Version of the document in the persistent data storage         |
+
+## Usage
+
+<<< ./snippets/update-java.java
+
+:::
+::: tab Kotlin
+
+## Arguments
+ 
+```kotlin
+  fun update(
+      index: String,
+      collection: String,
+      id: String?,
+      document: ConcurrentHashMap<String?, Any?>?,
+      waitForRefresh: Boolean? = null,
+      retryOnConflict: Int? = null,
+      source: Boolean? = null): CompletableFuture<ConcurrentHashMap<String, Any?>>
+```
+
+| Arguments          | Type                                         | Description                       |
+| ------------------ | -------------------------------------------- | --------------------------------- |
+| `index`            | <pre>String</pre>                            | Index                             |
+| `collection`       | <pre>String</pre>                            | Collection                        |
+| `id        `       | <pre>String</pre>                            | Document ID                        |
+| `document`         | <pre>ConcurrentHashMap<String, Any?></pre> | Partial document content |
+| `options`          | <pre>UpdateOptions</pre><br>(`null`)         | Optional parameters               |
```

Comment:
```suggestion
| `waitForRefresh`   | <pre>Boolean</pre>                           | If set to `true`, Kuzzle will wait for the persistence layer to finish indexing|
| `retryOnConflict`  | <pre>Int</pre>                           | The number of times the database layer should retry in case of version conflict |
| `source`           | <pre>Boolean</pre>                           | If true, returns the updated document inside the response |
```